src/Passes/CONTRIBUTING.md

```diff
@@ -0,0 +1,83 @@
+| --------------------- | -------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------- |
+| Namespace             | snake_case                       | Differentiates `namespace::ClassName` and `ClassName::SubClass` names                                                                         |
+| Class name            | CamelCase                        | To differentiate from STL types which ISO recommends (do not use "C" or "T" prefixes)                                                         |
+| Function name         | camelCase                        | Lower case start is almost universal except for .Net world                                                                                    |
+| Parameters/Locals     | snake_case                       | Vast majority of standards recommends this because \_ is more readable to C++ crowd (although not much to Java/.Net crowd)                    |
+| Member variables      | snake_case_with\_                | The prefix \_ is heavily discouraged as ISO has rules around reserving \_identifiers, so we recommend suffix instead                          |
+| Enums and its members | CamelCase                        | Most except very old standards agree with this one                                                                                            |
+| Globals               | g_snake_case                     | Avoid using globals whenever possible, but if you have to use `g_`.                                                                           |
+| Constants             | UPPER_CASE                       | Very contentious and we just have to pick one here, unless if is a private constant in class or method, then use naming for Members or Locals |
+| File names            | Match case of class name in file | Lot of pro and cons either way but this removes inconsistency in auto generated code (important for ROS)                                      |
+
+## Modernise when possible
+
+In general, modernise the code where possible. For instance, prefer `using` of `typedef`.
+
+## Header guards
+
+Prefer `#pragma once` over `#ifdef` protection.
+
+## Code TODOs must contain owner name or Github issue
+
+```sh
+% ./manage runci
+(...)
+QsPasses/src/OpsCounter/OpsCounter.cpp:39:21: error: missing username/bug in TODO [google-readability-todo,-warnings-as-errors]
+                    // TODO: Fails to load if this is present
+                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+                    // TODO(tfr): Fails to load if this is present
+```
+
+## Always add copyrights
+
+Always add copyrights at the top of the file.
+
+```text
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+```
+
+For header files, prefer to put `#prama once` before the copyright.
+
```

Comment:
Nit: `#prama once` does not relate to copyrights. Consider either removing or placing in a separate section.

Reply:
You are correct: `#pragma once` does not relate directly to the copyright. However, since all functions should have copyrights and all headers should have `#pragma once`, the suggestion is to always format your section like:
```
#pragma once
// Copyright ...
```
rather than
```
// Copyright ...
#pragma once
```
This is somewhat pedantic and I am happy to remove this.  The reason it is here is to add consistency to the files. Let me know what you think

Reply:
OK, makes sense.